Common/Securities/SecurityPortfolioManager.cs

```diff
@@ -368,30 +370,47 @@ public decimal TotalPortfolioValue
         {
             get
             {
-                // we can't include forex in this calculation since we would be double accounting with respect to the cash book
-                // we also exclude futures and CFD as they are calculated separately
-                decimal totalHoldingsValueWithoutForexCryptoFutureCfd = 0;
-                foreach (var kvp in Securities)
+                if (!_isTotalPortfolioValueValid)
                 {
-                    var position = kvp.Value;
-                    if (position.Type != SecurityType.Forex && position.Type != SecurityType.Crypto &&
-                        position.Type != SecurityType.Future && position.Type != SecurityType.Cfd)
+                    // we can't include forex in this calculation since we would be double accounting with respect to the cash book
+                    // we also exclude futures and CFD as they are calculated separately
+                    decimal totalHoldingsValueWithoutForexCryptoFutureCfd = 0;
+                    foreach (var kvp in Securities)
                     {
-                        totalHoldingsValueWithoutForexCryptoFutureCfd += position.Holdings.HoldingsValue;
+                        var position = kvp.Value;
+                        // we can't include forex in this calculation since we would be double accounting with respect to the cash book
+                        // we also exclude futures and CFD as they are calculated separately
+                        if (position.Type != SecurityType.Forex && position.Type != SecurityType.Crypto &&
+                            position.Type != SecurityType.Future && position.Type != SecurityType.Cfd)
+                        {
+                            totalHoldingsValueWithoutForexCryptoFutureCfd += position.Holdings.HoldingsValue;
+                        }
                     }
-                }
 
-                var totalFuturesAndCfdHoldingsValue = Securities
-                    .Where(x => x.Value.Type == SecurityType.Future || x.Value.Type == SecurityType.Cfd)
-                    .Aggregate(0m, (d, pair) => d + pair.Value.Holdings.UnrealizedProfit);
+                    var totalFuturesAndCfdHoldingsValue = Securities
+                        .Where(x => x.Value.Type == SecurityType.Future || x.Value.Type == SecurityType.Cfd)
+                        .Aggregate(0m, (d, pair) => d + pair.Value.Holdings.UnrealizedProfit);
```

Comment:
We can roll this `Aggregate` call up into the loop above

Reply:
👍  Addressed in the previous PR https://github.com/QuantConnect/Lean/pull/3271/files#diff-ea81a360172dd59a9b5b98d829e4f26fR387